admin/index_m.js

```diff
@@ -0,0 +1,111 @@
+// This will be called by the admin adapter when the settings page loads
+function load(settings, onChange) {
+    // example: select elements with id=key and class=value and insert value
+    if (!settings) return;
+    $('.value').each(function () {
+        let $key = $(this);
+        let id = $key.attr('id');
+        if ($key.attr('type') === 'checkbox') {
+            // do not call onChange direct, because onChange could expect some arguments
+            $key.prop('checked', settings[id])
+                .on('change', () => onChange());
+        } else {
+            // do not call onChange direct, because onChange could expect some arguments
+            $key.val(settings[id])
+                .on('change', () => onChange())
+                .on('keyup', () => onChange());
+        }
+    });
+    onChange(false);
+    // reinitialize all the Materialize labels on the page if you are dynamically adding inputs:
+    if (M) M.updateTextFields();
+
+    // Selection Box added from https://forum.iobroker.net/topic/10186/adapterentwicklung-object-id-baum/6
+
+    $('#solarPowerForeignSearch').click(function () {
+        initSelectId(function (sid) {
+            sid.selectId('show', $('#solarPowerForeignObjectID').val(), function (newId) {
+                if (newId != $('#solarPowerForeignObjectID').val()) {
+                    $('#solarPowerForeignObjectID').val(newId);
+                    $("#solarPowerLbl").addClass("active ");
+                    onChange();
+                }
+            });
+        });
+    });
+    $('#houseConsumptionForeignSearch').click(function () {
+        initSelectId(function (sid) {
+            sid.selectId('show', $('#houseConsumptionForeignObjectID').val(), function (newId) {
+                if (newId != $('#houseConsumptionForeignObjectID').val()) {
+                    $('#houseConsumptionForeignObjectID').val(newId);
+                    $("#houseConsumptionLbl").addClass("active ");
+                    onChange();
+                }
+            });
+        });
+    });
+    $('#houseBatteryForeignSearch').click(function () {
+        initSelectId(function (sid) {
+            sid.selectId('show', $('#houseBatteryForeignObjectID').val(), function (newId) {
+                if (newId != $('#houseBatteryForeignObjectID').val()) {
+                    $('#houseBatteryForeignObjectID').val(newId);
+                    $("#houseBatteryLbl").addClass("active ");
+                    onChange();
+                }
+            });
+        });
+    });
+}
+
+// This will be called by the admin adapter when the user presses the save button
+function save(callback) {
+    // example: select elements with class=value and build settings object
+    var obj = {};
+    $('.value').each(function () {
+        var $this = $(this);
+        if ($this.attr('type') === 'checkbox') {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MK-2001/ioBroker.go-e/pullRequest?prid=7592349)
```suggestion
        if ($this.attr("type") === 'checkbox') {
```